tools/meson-vcs-tag.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/sh
+
+set -eu
+set -o pipefail
+
+dir="$1"
+tag="$2"
+fallback="$3"
+file="${4:-}"
+
+if [ -z "$tag" ]; then
+        # Apparently git describe has a bug where it always considers the work-tree
+        # dirty when invoked with --git-dir (even though 'git status' is happy). Work
+        # around this issue by cd-ing to the source directory.
+        tag="$(cd "$dir" && git describe --dirty=+ 2>/dev/null | sed 's/^v//' || echo "$fallback")"
```

Comment:
I typically recommend using an explicit `--abbrev=` as part of the `git describe` command line. (I also recommend something like 12 or 14 for abbrev length.)

For instance, I always have something like `git config core.abbrev 12` or `git config core.abbrev 14` set on my workdirs, to keep the length consistent, but others might not... So making sure the build system will always produce consistent naming regardless of local git config would be great.

Reply:
That's a good point.